mergify_engine/rules/__init__.py

```diff
@@ -482,18 +516,62 @@ async def get_mergify_config_content(
 
 
 class MergifyConfig(typing.TypedDict):
+    defaults: Defaults
     pull_request_rules: PullRequestRules
     queue_rules: QueueRules
 
 
+def merge_config(config: typing.Any) -> typing.Any:
+    pull_request_rules = config["pull_request_rules"]
+
+    if defaults := config.get("defaults"):
+        if defaults_actions := defaults.get("actions"):
+            for i, rule in enumerate(config["pull_request_rules"]):
+                actions = rule["actions"]
+
+                for action_name in list(actions):
+                    if action_name not in list(defaults_actions):
+                        continue
+
+                    merged_action = defaults_actions[action_name] | actions[action_name]
+                    pull_request_rules[i]["actions"][action_name] = (
+                        pull_request_rules[i]["actions"][action_name] | merged_action
+                    )
+    return pull_request_rules
+
+
 async def get_mergify_config(
     redis_cache: utils.RedisCache,
     client: github.AsyncGithubInstallationClient,
     repo: github_types.GitHubRepository,
     ref: typing.Optional[github_types.GitHubRefType] = None,
 ) -> typing.Tuple[str, MergifyConfig]:
     filename, content = await get_mergify_config_content(redis_cache, client, repo, ref)
+
     try:
-        return filename, typing.cast(MergifyConfig, UserConfigurationSchema(content))
+        config = YamlSchema(content)
+    except voluptuous.Invalid as e:
+        raise InvalidRules(e, filename)
+
+    try:
+        partial_config = config.copy()
+        if partial_config.get("queue_rules"):
+            partial_config.pop("queue_rules")
+
+        PartialUserConfigurationValidator(partial_config)
```

Comment:
I think you can just validate everything here.

```UserConfigurationSchema(config)```

Or maybe I miss something.